meilisearch/task.py

```diff
@@ -0,0 +1,116 @@
+from time import sleep
+from datetime import datetime
+from typing import Any, Dict, List, Optional
+
+from meilisearch._httprequests import HttpRequests
+from meilisearch.config import Config
+from meilisearch.errors import MeiliSearchTimeoutError
+
+class Task:
+    """
+    Tasks routes wrapper.
+
+    Task class gives access to all Task routes and child routes via an Index (inherited).
+    https://docs.meilisearch.com/reference/api/task.html
+    """
+
+    def __init__(
+        self,
+        config: Config,
+        uid: Optional[str] = None,
+    ) -> None:
+        """
+        Parameters
+        ----------
+        config:
+            Config object containing permission and location of MeiliSearch.
+        """
+        self.config = config
+        self.http = HttpRequests(config)
+        self.uid = uid
+
+    def get_tasks(self) -> Dict[str, List[Dict[str, Any]]]:
+        """Get all tasks.
+
+        Returns
+        -------
+        task:
+            Dictionary containing a list of all enqueued, processing, succeeded or failed tasks.
+
+        Raises
+        ------
+        MeiliSearchApiError
+            An error containing details about why MeiliSearch can't process your request. MeiliSearch error codes are described here: https://docs.meilisearch.com/errors/#meilisearch-errors
+        """
+        if self.uid is None:
+            return self.http.get(
+                f'{self.config.paths.task}'
+            )
+        return self.http.get(
+            f'{self.config.paths.index}/{self.uid}/{self.config.paths.task}'
+        )
+
+
+    def get_task(self, uid: int) -> Dict[str, Any]:
+        """Get one task.
+
+        Parameters
+        ----------
+        uid:
+            Identifier of the task.
+
+        Returns
+        -------
+        task:
+            Dictionary containing information about the processed asynchronous task.
+
+        Raises
+        ------
+        MeiliSearchApiError
+            An error containing details about why MeiliSearch can't process your request. MeiliSearch error codes are described here: https://docs.meilisearch.com/errors/#meilisearch-errors
+        """
+        if self.uid is None:
+            return self.http.get(
+                f'{self.config.paths.task}/{uid}'
+            )
+        return self.http.get(
+            f'{self.config.paths.index}/{self.uid}/{self.config.paths.task}/{uid}'
```

Comment:
Same comment as with tasks. I think this should be

```py
return self.http.get(f'{self.config.paths.task}/{uid}')
```

Reply:
Sorry, I see you can do it both ways so the index id is needed

Reply:
It makes no difference but we prefer to make both methods accessible.

Reply:
Making it assessable from the index I understand. What I am saying is having the function calculate the different urls is extra work that isn’t needed. The user isn’t involved in, and doesn’t see this part so they can still use the function from within the index even if the index I’d isn’t sent in the url.

Hopefully this is more clear, but I have a feeling I am still not explaining it well.

Reply:
No, it's very clear. We had a meeting about the new implementation and I don't remember why we decided to implement it as it was. I will leave it as it is and put the subject back with the integration team.
Because you are right the user going to have the same result maybe it's to be sure this task belongs to a specific index otherwise the methods will fail.

Reply:
Or maybe I don't understand well how we should implement it. Anyway, I'm going to find out for sure before the final merge of the bump.